eclipse-collections/src/main/java/org/eclipse/collections/impl/map/sorted/immutable/ImmutableTreeMap.java

```diff
@@ -231,7 +235,7 @@ public K lastKey()
     }
 
     @Override
-    public Set<K> keySet()
+    public ImmutableSortedMapKeySet keySet()
```

Comment:
@johnjstuart I don't think it's a good idea to publicly expose a class internal to the file here, let's keep the interface, no?